### PR TITLE
Add set timer script, restart timer, disable timer events

### PR DESCRIPTION
**Pull request by gregtour**
* **Please check if the PR fulfills these requirements**
- [X] The commit message follows our [guidelines](/chrismaltby/gb-studio/blob/develop/.github/COMMIT_MESSAGE_GUIDELINES.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This PR introduces a timer script feature.

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

See below.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

* **Documentation**:

**Timer: Set timer script**
SetTimerScript

Execute the specified script after a fixed amount of time. The timer will repeat until it is disabled or the scene changes. The timer does not count down when a script is running or dialogue is open. The timer duration is between 0.5 seconds and 60 seconds. There is only one timer script at a time.

**Timer: Restart timer**
RestartTimer

Restart the timer countdown.

**Timer: Disable timer script**
DisableTimer
Remove the timer and disable the timer script. The timer is automatically removed when the scene changes.

### CODE REVIEW

**Comment by chrismaltby** [comment]
Interesting, I like it!

Is there any chance you can add a minimal sample project `Test_TimerScript` into `test/projects` it will just a bit easier for me to test out on the different builds before merging. 

Just to check have you been running this one on Windows, Mac or Linux? Thanks

**Comment by gregtour** [comment]
I added an example. I've been developing on macOS but just took some time to test the example on Linux and Windows. Works fine on Linux but it can cause a crash to a white screen in the game on Windows.

File: src/lang/en.json [hunk 1]
```diff
@@ -156,6 +156,11 @@
   "FIELD_FLAG_6": "Flag 6",
   "FIELD_FLAG_7": "Flag 7",
   "FIELD_FLAG_8": "Flag 8",
+  "FIELD_SET_TIMER": "Add repeating timer script",
```
Comment (pau-tomas) [inline src/lang/en.json]: NIT: This should probably say "Set" rather than "Add" as there's at max one timer executing at any given time.

Comment (gregtour) [reply src/lang/en.json]: Fixed

**Comment by chrismaltby** [comment]
Finally got some time to check this one, test project is working great! Thanks!

**Pull request merged**